server/src/test/java/io/druid/server/coordinator/DruidCoordinatorTest.java

```diff
@@ -342,6 +342,9 @@ public void childEvent(
       Thread.sleep(50);
     }
 
+    // wait historical data to be updated
+    Thread.sleep(druidCoordinatorConfig.getCoordinatorPeriod().getMillis());
```

Comment:
but I think we'll still see sporadic failures unless there's more of a check in place. Can we add some sort of conditional check?